### ISSUE TITLE
Some requests times out in http client connection pool

### ISSUE THREAD

**Issue by luckyswede** [micronaut-projects/micronaut-core#1503]
When http client connection pooling is enabled we consistently see read timeout errors.
Unfortunately I don't have a unit test that reproduces this, but it's easily reproducable using a simple app and running eg `ab`.

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce
Follow the readme in the example app

### Expected Behaviour

No timeouts should happen.

### Actual Behaviour

Some requests times out with this exception: 
```
11:30:36.924 [RxComputationThreadPool-1] ERROR i.m.h.s.netty.RoutingInBoundHandler - Unexpected error occurred: Read Timeout
io.micronaut.http.client.exceptions.ReadTimeoutException: Read Timeout
	at io.micronaut.http.client.exceptions.ReadTimeoutException.<clinit>(ReadTimeoutException.java:26)
	at io.micronaut.http.client.DefaultHttpClient.lambda$null$26(DefaultHttpClient.java:1070)
	at io.reactivex.internal.operators.flowable.FlowableOnErrorNext$OnErrorNextSubscriber.onError(FlowableOnErrorNext.java:103)
	at io.micronaut.reactive.rxjava2.InstrumentedSubscriber.lambda$onError$1(InstrumentedSubscriber.java:96)
	at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:52)
	at io.micronaut.http.context.ServerRequestContext.lambda$instrument$0(ServerRequestContext.java:68)
	at io.micronaut.reactive.rxjava2.InstrumentedSubscriber.onError(InstrumentedSubscriber.java:100)
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutSubscriber.onTimeout(FlowableTimeoutTimed.java:139)
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutTask.run(FlowableTimeoutTimed.java:170)
	at io.micronaut.http.context.ServerRequestContext.with(ServerRequestContext.java:52)
	at io.micronaut.http.context.ServerRequestContext.lambda$instrument$0(ServerRequestContext.java:68)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### Environment Information

- **Operating System**: Linux, MacOS
- **Micronaut Version:** 1.1.0.RC2
- **JDK Version:** 11

### Example Application

https://github.com/luckyswede/micronaut-clientpooltest

**Comment by graemerocher** [comment]
So looking at how other libraries handle connection pooling with Netty it seems that using `ReadTimeoutHandler` is problematic when combined with `ChannelPool` since `ChannelPool` maintains stateful connections.

We already apply read timeout to the reactive flow here, essentially applying it twice https://github.com/micronaut-projects/micronaut-core/blob/master/http-client/src/main/java/io/micronaut/http/client/DefaultHttpClient.java#L1059 so I think what should be done here is to note use `ReadTimeoutHandler` in the connection pool case and rely on a higher level read timeout applied to the Flowable.

**Comment by graemerocher** [comment]
If you wish to verify the fix you can do so by adding the snapshot repo to `build.gradle`:

```groovy
repositories {
    ...
    maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
}
```

And specifying the snapshot version as your BOM:

```groovy
dependencyManagement {
    imports {
        mavenBom 'io.micronaut:micronaut-bom:1.1.0.BUILD-SNAPSHOT'
    }
}
```

**Comment by graemerocher** [comment]
I found one last issue in testing related to when `Connection: close` is sent from the server. The above commit resolves this case as well.

**Comment by luckyswede** [comment]
Cool, I've tested your fix using `fortio` as well (HTTP/1.1, with a few permutations of the `connection` header).
Seems to work.
